src/assets/JourneyWithGo-ABlogSeries/goFundamentals.md

```diff
@@ -0,0 +1,821 @@
+---
+title: Fundamentals of Go - FARMISS
+authorName: Prarup Gurung (प्रारुप गुरुङ्ग)
+authorAvatar: https://avatars.githubusercontent.com/u/8559121?v=4
+authorLink: https://github.com/grgprarup
+createdAt: Oct 26, 2023
+tags: function, array, rune, method, interface, string, struct
+banner: https://blog.jankaritech.com/src/assets/JourneyWithGo-ABlogSeries/images/control-statements-banner.svg
+seriesTitle: Journey With Go - A Blog Series
+episode: 3
+---
+
+**[गो](https://go.dev "Go in Nepali")** **"Journey With Go - A Blog Series"** about the basics of Go `Go Basic`,
+advanced concepts in Go `Go Beyond Basics`, testing with go `Godog as Test Framework` and many more.
+
+This is the third installment of the series. You are here which means you have already read the first and second blog of
+the series. If you haven't, please read them
+first [Go Basics](https://blog.jankaritech.com/#/blog/Journey%20With%20Go%20-%20A%20Blog%20Series/Go%20Basics%20-%20The%20Starting%20Point "Go Basics"), [Control Statements](https://blog.jankaritech.com/#/blog/Journey%20With%20Go%20-%20A%20Blog%20Series/Control%20Statements%20-%20Control%20Your%20Go "Control Statements").
+
+In this blog, we will learn about the `Function`, `Array`, `Rune`, `Method`, `Interface`, `String`, and `Struct` (FARMISS) in
+Go. **So, let's get started.**
+
+## **1. Rune**
+
+In Go, a rune is a built-in type that represents a Unicode code point. A rune literal is expressed as one or more
+characters enclosed in single quotes, as in 'x' or '\n'.
+Each Unicode code point corresponds to a unique character, which can be a letter, a numeral, a symbol, or an emoji.
+
+```go
+package main
+
+import (
+  "fmt"
+)
+
+func main() {
+  var om rune = 'ॐ'       // rune literal declared using `rune` keyword, use single quotes
+  fmt.Println(om)         // 2384, which is the Unicode code point for the Om symbol
+  fmt.Printf("%T\n", om)  // int32, which is the underlying type of rune
+  fmt.Println(string(om)) // ॐ, which is the string representation of the Om symbol
+
+  // more examples
+  var heart rune = '♥'
+  fmt.Println(heart)         // 9829
+  fmt.Println(string(heart)) // ♥
+
+  var smiley rune = '😀'
+  fmt.Println(smiley)         // 128512
+  fmt.Println(string(smiley)) // 😀
+}
+```
+
+It is particularly important when dealing with text in different languages or when working with special symbols and
+emojis. It ensures that each character is treated as a single entity, regardless of its complexity or language.
+
+## **2. String**
+
+In Go, a string is a sequence of bytes (not characters) that is used to represent text. Strings are immutable, which
+means once created, their values cannot be changed. However, you can assign a new value to a string variable, which will
+create a new string.
+
+```go
```

Comment:
A Note stating the meaning of `' '` (rune) and `" "` (string) would be great here. like using `'` is a rune and using `"` is a string.